engine/src/main/java/com/datatorrent/stram/client/StramClientUtils.java

```diff
@@ -118,6 +118,8 @@
   public static final long DELEGATION_TOKEN_MAX_LIFETIME_DEFAULT = 7 * 24 * 60 * 60 * 1000;
   public static final String TOKEN_REFRESH_PRINCIPAL = StramUserLogin.DT_AUTH_PREFIX + "token.refresh.principal";
   public static final String TOKEN_REFRESH_KEYTAB = StramUserLogin.DT_AUTH_PREFIX + "token.refresh.keytab";
+  public static final String CUSTOM_SSL_KEYSTORE_CONFIG = StreamingApplication.APEX_PREFIX + "ssl.keystore.config";
```

Comment:
@vrozov  STRAM_HTTP_CUSTOM_CONFIG is an attribute because only the StreamingAppMasterService uses it and that is the most common way to pass values (attributes) to it.

CUSTOM_SSL_KEYSTORE_CONFIG is a setting because the Apex CLI (StramClient) implements it and the most common way to pass settings to it are via dt-site.xml or thru -D option on the launch command.

The code in this JIRA is going to use the feature added in APEXCORE-711. We had designed it to be like this based on user+developer discussion captured in http://apache-apex-developers-list.78491.x6.nabble.com/Enhancement-to-support-custom-SSL-configuration-td14335.html . So there is no question of the client setting blindly overwriting the server setting